eth_account/account_local_actions.py

```diff
@@ -0,0 +1,73 @@
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+from eth_keyfile.keyfile import (
+    KDFType,
+)
+from eth_keys.datatypes import (
+    PrivateKey,
+)
+from eth_typing import (
+    HexStr,
+)
+from eth_utils.curried import (
+    combomethod,
+)
+
+from eth_account.datastructures import (
+    SignedMessage,
+    SignedTransaction,
+)
+from eth_account.messages import (
+    SignableMessage,
+)
+
+
+class AccountLocalActions(ABC):
```

Comment:
I just noticed that there is already an ABC called BaseAccount. I wonder if we could use that instead of this new ABC? It looks like they're pretty similar.

Reply:
That's what the `LocalAccount` inherits from. `BaseAccount` and `LocalAccount` are separate entities from `Account` - an instantiated `LocalAccount` has an `Account` as an attribute, and an `Account` can generate new instances of `LocalAccount`. I think it's pretty confusing. But that's why we need something separate.

Reply:
I was playing around with this a little, and I think it makes more sense just to keep Account separate, i.e. not inheriting from `AccountLocalActions`. I'm not sure what we gain from the `AccountLocalActions` abstract class. If we remove it, then `Account` can be its own separate entity and we don't need `AccountLocalActions` any more. What do you think about that? I may be missing something too.

Reply:
It's the circular import - `LocalAccount` needs an `Account` as for its `_public_api`, but an Account needs to be able to return instances of `LocalAccount`

Reply:
The other idea was to split Account into 2 classes - signing methods and creation methods, with the full method implementation in each. Those could be joined into what is currently `Account`, and `LocalAccount` would just take the signing one.

Reply:
> It's the circular import - LocalAccount needs an Account as for its _public_api, but an Account needs to be able to return instances of LocalAccount

Ugh, just saw that. Sorry for the goose chase 🪿 . I think this is okay for now. Will continue reviewing!